qt/python/mantidqt/project/project.py

```diff
@@ -0,0 +1,138 @@
+            project_saver.save_project(directory=self.last_project_location, workspace_to_save=workspaces_to_save,
+                                       interfaces_to_save=None)
+            self.__saved = True
+
+    def save_as(self):
+        directory = self._get_directory_finder(accept_mode=QFileDialog.AcceptSave)
+
+        # If none then the user cancelled
+        if directory is None:
+            return
+
+        # todo: get a list of workspaces but to be implemented on GUI implementation
+        self.last_project_location = directory
+        workspaces_to_save = AnalysisDataService.getObjectNames()
+        project_saver = ProjectSaver(self.project_file_ext)
+        project_saver.save_project(directory=directory, workspace_to_save=workspaces_to_save, interfaces_to_save=None)
+        self.__saved = True
+
+    @staticmethod
+    def _get_directory_finder(accept_mode):
+        directory = None
+        # Check if it exists
+        first_pass = True
+        while first_pass or (not os.path.exists(directory) and os.path.exists(directory + (os.path.basename(directory)
+                                                                                           + ".mtdproj"))):
+            first_pass = False
+            directory = open_a_file_dialog(accept_mode=accept_mode, file_mode=QFileDialog.DirectoryOnly)
+            if directory is None:
+                # Cancel close dialogs
+                return
+
+        return directory
+
+    def load(self):
+        directory = self._get_directory_finder(accept_mode=QFileDialog.AcceptOpen)
+
+        # If none then the user cancelled
+        if directory is None:
+            return
+
+        project_loader = ProjectLoader(self.project_file_ext)
+        project_loader.load_project(directory)
+        self.last_project_location = directory
+
+    def offer_save(self, parent):
+        """
+        :param parent: QWidget; Parent of the QMessageBox that is popped up
+        :return: Bool; Returns false if no save needed/save complete. Returns True if need to cancel closing.
+        """
+        # If the current project is saved then return and don't do anything
+        if self.__saved:
+            return
+
+        result = self._offer_save_message_box(parent)
+
+        if result == QMessageBox.Yes:
+            self.save()
+        elif result == QMessageBox.Cancel:
+            return True
+        # else
+        return False
+
+    @staticmethod
+    def _offer_save_message_box(parent):
+        return QMessageBox.question(parent, 'Unsaved Project', "The project is currently unsaved would you like to "
+                                    "save before closing?", QMessageBox.Yes | QMessageBox.No | QMessageBox.Cancel,
+                                    QMessageBox.Yes)
+
+    def modified_project(self):
+        self.__saved = False
+
+    def anyChangeHandle(self):
+        self.modified_project()
+
+    def __get_saved(self):
```

Comment:
Can we move the definition of 

```python
    def __get_saved(self):
        return self.__saved
    saved = property(__get_saved)
```
to just under `__init__.py`. It's clearer one properties are available then.

Reply:
Moved :+1: